### ISSUE TITLE
Simplify the definition of the tabular data model?

### ISSUE THREAD

**Issue by iherman** [w3c/csvw#68]
The current document still describes both the core and the tabular data model using annotated rows and annotated cells as separate entities. We agreed at TPAC to remove the row and cell annotation from the metadata file; do we still want to keep these notions in the model? I would think that consistency should mean not to keep them.

**Comment by JeniT** [comment]
We agreed at TPAC not to specify include properties for rows or cells within the _schemas_ but there are still annotations (eg notes) on rows and columns, so the model still needs to enable annotations on both.

**Comment by iherman** [comment]
> On 29 Nov 2014, at 22:55 , Jeni Tennison notifications@github.com wrote:
> 
> We agreed at TPAC not to specify include properties for rows or cells within the schemas but there are still annotations (eg notes) on rows and columns, so the model still needs to enable annotations on both.

Can't we use the RFC on CSV Fragments for that purpose? An notes (annotations) does have the notion of targets, and that could be used...

Ivan

> —
> Reply to this email directly or view it on GitHub.

---

Ivan Herman, W3C
Digital Publishing Activity Lead
Home: http://www.w3.org/People/Ivan/
mobile: +31-641044153
ORCID ID: http://orcid.org/0000-0003-0782-2704

**Comment by JeniT** [comment]
Yes, fragment identifiers could be used to work out which rows/cells are getting annotations. Or there might be other methods of identifying them. The conceptual model of the table needs to include annotations on rows and cells either way.

**Comment by 6a6d74** [comment]
Question: for annotations (and other stuff) are we annotating the CSV data file _or_ the abstract data model.

This is particularly relevant for rows ...

Example:

```
GID,On Street,Species
1,ADDISON AV,Celtis australis
2,EMERSON ST,Liquidambar styraciflua
```

RFC 7111 has the row number beginning with the first line in the file, with the first row at position 1 - so `#row=2` identifies `1,ADDISON AV,Celtis australis` ... 

where as the Tabular Data Model suggests that position 1 starts with the _data_, in which case `#row=2` identifies `2,EMERSON ST,Liquidambar styraciflua`. [I'm not 100% sure this is true - I can't find any specific details in the doc]

(note that RFC 7111 _strictly_ only applies to `text/csv` so wouldn't apply to tab-separated-value formatted files ...)

Personally, I think it would be safer to stick with the RFC 7111 fragment identifiers if we're referencing rows, columns, cells etc. in order to avoid confusion - and for us to say that "this applies to _all_ tabular data file formats". This would allow us to annotate the _CSV file_ itself.

If people want to annotate the object described by, say, a given row then they might chose to use the URL for that object in the annotation as opposed to the row fragment identifier.

**Comment by iherman** [comment]
This relates to issue #32, see [my reply this morning](https://github.com/w3c/csvw/issues/32#issuecomment-65037619). I think that what I replied this morning amounts to saying that we use RFC7111 in terms of the CSV file itself. Isn't it what RFC7111 has been defined for in the first place?

**Comment by 6a6d74** [comment]
Thanks & apologies for x-posting ...

**Comment by gkellogg** [comment]
> Personally, I think it would be safer to stick with the RFC 7111 fragment identifiers if we're referencing rows, columns, cells etc. in order to avoid confusion - and for us to say that "this applies to all tabular data file formats". This would allow us to annotate the CSV file itself.

+1, I agree we should stick with RFC 7111 row numbering.

**Comment by 6a6d74** [comment]
In the [metadata vocab doc, section 3.1 Property Syntax](http://w3c.github.io/csvw/metadata/#property-syntax) we identify a special `_row` variable for use un URI Templates.

Re-using the example from above ...

```
GID,On Street,Species
1,ADDISON AV,Celtis australis
2,EMERSON ST,Liquidambar styraciflua
```

If we stick with RFC 7111 numbering, then I would expect that when processing row `1,ADDISON AV,Celtis australis` then the URI Template `{_row}` would evaluate to 2.

This is at variance with @JeniT's comment in issue #32:

"""
The tabular data model doesn't include a header row, only data rows, and the numbering of these start from 1. When a CSV file that includes a header row gets parsed into a tabular data model, the headers become annotations on the columns, not a separate row.
""" 

I can see merits of both approaches. I am still not 100% clear - even given @iherman's comment in #32.

**Comment by iherman** [comment]
Yep, this seems to be messy:-( There is a discrepancy among the row number in the model and in the original CSV file. What seems to happen is that
- for templating and conversion results the data model counts, ie, (in your example) the data row should be '1'
- for fragments what counts is the CSV file, ie, for that purpose the data line has row = 2

And we can bet that this will lead to mistakes. So maybe the metadata should work only on the CSV row numbers? That makes it clear/clean, although, conceptually, this means the metadata is not on the data model but on the CSV file. Arghh...

Ivan

> On 02 Dec 2014, at 10:57 , Jeremy Tandy notifications@github.com wrote:
> 
> In the metadata vocab doc, section 3.1 Property Syntax we identify a special _row variable for use un URI Templates.
> 
> Re-using the example from above ...
> 
> GID,On Street,Species
> 1,ADDISON AV,Celtis australis
> 2,EMERSON ST,Liquidambar styraciflua
> 
> If we stick with RFC 7111 numbering, then I would expect that when processing row 1,ADDISON AV,Celtis australis then the URI Template {_row} would evaluate to 2.
> 
> This is at variance with @JeniT's comment in issue #32:
> 
> """
> The tabular data model doesn't include a header row, only data rows, and the numbering of these start from 1. When a CSV file that includes a header row gets parsed into a tabular data model, the headers become annotations on the columns, not a separate row.
> """
> 
> I can see merits of both approaches. I am still not 100% clear - even given @iherman's comment in #32.
> 
> —
> Reply to this email directly or view it on GitHub.

---

Ivan Herman, W3C
Digital Publishing Activity Lead
Home: http://www.w3.org/People/Ivan/
mobile: +31-641044153
ORCID ID: http://orcid.org/0000-0003-0782-2704

**Comment by 6a6d74** [comment]
It's the confusion between approaches that worries me the most ... Have added a "discuss @ teleconf" label.

**Comment by danbri** [comment]
See http://www.w3.org/2015/02/12-csvw-irc#T15-47-08

**Comment by JeniT** [comment]
Discussed at Feb F2F: this will be resolved through resolution of #32 and #24 and #71
